src/Modules/RoomsModule/components/RoomsData/RoomsData.tsx

```diff
@@ -1,7 +1,283 @@
-import React from 'react'
+import {
+  Container,
+  FormControl,
+  Grid,
+  InputLabel,
+  Select,
+  TextField,
+  MenuItem,
+  Button,
+  Box,
+  Checkbox,
+  ListItemText,
+  Alert,
+  CircularProgress,
+} from "@mui/material";
+import CloudUploadIcon from "@mui/icons-material/CloudUpload";
+import { SubmitHandler, useForm } from "react-hook-form";
+import { FacilitiesProps, Inputs } from "../../../../interfaces/interface";
+import { userRequest } from "../../../../utils/request";
+import { useEffect, useRef, useState } from "react";
+import { getErrorMessage } from "../../../../utils/error";
+import { useToast } from "../../../Context/ToastContext";
+import { Link, useLocation, useNavigate } from "react-router-dom";
 
 export default function RoomsData() {
+  const [facilitiesList, setFacilitiesList] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [images, setImages] = useState<{ file: File; previewUrl: string }[]>(
+    []
+  );
+  const [imgUpload, setImgUpload] = useState<FileList | null | []>([]);
+
+  const location = useLocation();
+  const roomId = location.pathname.split("/")[3];
+  const roomData = location.state?.roomData;
+  const state = location.state?.state;
+  const timeoutRef = useRef<number | null>(null);
+  const navigate = useNavigate();
+  const { showToast } = useToast();
+  const {
+    register,
+    handleSubmit,
+    watch,
+    setValue,
+    formState: { errors },
+  } = useForm<Inputs>();
+
+  const handleImageUpload = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const files = e.target.files;
+    setImgUpload(files);
+    if (files) {
+      const newImages: { file: File; previewUrl: string }[] = Array.from(
+        files
+      ).map((file) => ({
+        file,
+        previewUrl: URL.createObjectURL(file),
```

Comment:
We need to revoke these object on a useEffect when the component unmount (in the useEffect's return) to avoid having many images  which are not  used any more in the memory.

https://developer.mozilla.org/en-US/docs/Web/API/URL/revokeObjectURL_static